### PR TITLE
[ETL-385] Add in s3 production data pre-etl bucket

### PR DESCRIPTION
**Pull request by thomasyu888**
Purpose: This PR create the production data pre-etl bucket which the ETL pipeline will read from.

This bucket is to make sure we have a data source that is constant and cannot be edited by externals. Only Sage has access to this bucket, and this will be linked to the RECOVER Synapse project (syn51105296). The file list will be replicated in Synapse via STS token

### CODE REVIEW

File: config/prod/s3-production-data.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# This is the pre-ETL bucket (SYNAPSE_ATTACHED_BUCKET)
+template:
+  type: file
+  path: s3-bucket.yaml
+stack_name: recover-production-input-data
```
Comment (rxu17) [inline config/prod/s3-production-data.yaml:5]:
```suggestion
stack_name: recover-input-bucket
```

File: config/prod/s3-production-data.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+dependencies:
+  - prod/cfn-s3objects-macro.yaml
+parameters:
+  BucketName: recover-production-input-data
```
Comment (rxu17) [inline config/prod/s3-production-data.yaml:9]:
```suggestion
  BucketName: recover-input-data
```

File: config/prod/s3-production-data.yaml
Comment (rxu17) [inline config/prod/s3-production-data.yaml:1]: NIT: Would also rename file from `config/prod/s3-production-data.yaml` to `config/prod/s3-input-bucket.yaml`

**Review by rxu17 — APPROVED** [review]
LGTM! Just some minor comments, thanks!

File: config/prod/s3-production-data.yaml
Comment (thomasyu888) [reply config/prod/s3-production-data.yaml:1]: ha, the branch name got me all confused, but yes this is the pre-ETL bucket. 

Thanks for the thorough review!  I made all the changes

**Pull request merged**